lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex

```
@@ -30,6 +27,24 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
       {:error, "Can't calculate network growth"}
   end
 
+  def mvrv_ratio(_root, args, _resolution) do
+    with {:ok, contract, token_decimals} <- Project.contract_info_by_slug(args.slug),
+         {:ok, mvrv_ratio} <-
+           Mvrv.mvrv_ratio(contract, args.slug, args.from, args.to, args.interval, token_decimals) do
+      {:ok, mvrv_ratio}
+    else
+      error ->
+        Logger.error("Can't calculate MVRV ratio. Reason: #{inspect(error)}")
+
+        {:error, "Can't calculate MVRV ration"}
+    end
+  rescue
```

Let's rescue only exceptions for functions that are used in `all_*` queries because otherwise they'll fail the whole query. In that case it's ok to let it crash